test/utils.test.ts

```diff
@@ -1014,8 +1014,6 @@ describe("utils", () => {
             vscodeMock.workspace.getConfiguration.mockImplementation(() => {
                 return {
                     ...originalWorkspaceMockConfig,
-                    fillerPropertyOne: 'Need to add filler values since the extension related telemetry functions removes the first 4 entries to create an array of extension settings',
-                    fillerPropertyTwo: 'The first 4 entries for a workspaceConfiguration object are class related functions, the rest are all extension related settings.',

```

Comment:
are these changes required for this PR

Reply:
These actually somewhat related to make the [reportExtensionSettings test](https://github.com/microsoft/vscode-edge-devtools/blob/52d9d1896f28f78b6257740c33383e5324e93024/test/utils.test.ts#L1027) work after adding `update` [calls](https://github.com/microsoft/vscode-edge-devtools/blob/aac3c68f4b7cd343ed0385b66fa9680848f2acfd/src/extension.ts#L182) in `extension.ts`

Kind of convoluted, but here is the explanation if you care to read:

Basically  `reportExtensionSettings` is a function that grabs all settings and their values from the workspace config object via `Object.entries(vscode.workspace.getConfiguration(SETTINGS_STORE_NAME))`.  The first four entries in the array generated by `Object.entries` are the methods that the config object has which we aren't interested in. In the `reportExtensionSettings` function, [we splice the first four indices away](https://github.com/microsoft/vscode-edge-devtools/blob/main/src/utils.ts#L682) and the remaining array contains all the settings information.

I initially added these filler properties since we only [mocked 2](https://github.com/microsoft/vscode-edge-devtools/pull/462/files#diff-1ba4fa139870853d3020cb0394b69c4f19e55cd6da45d8442bff47e17e7dee94L82-R111) of the 4 methods of `getConfiguration`. Since the `reportExtensionSettings` function spliced away 4 indices, in the test environment, it would splice away 2 of the mocked functions and then 2 settings objects.  I fixed this by adding filler properties to the `getConfiguration` call so that it would splice the 2 mocked methods and the 2 filler properties, leaving all the settings objects intact.

I needed to mock `config.update` after adding `update` [calls](https://github.com/microsoft/vscode-edge-devtools/blob/aac3c68f4b7cd343ed0385b66fa9680848f2acfd/src/extension.ts#L182) in `extension.ts` and realized the obvious solution would be to mock the remaining 2 functions instead of having these filler properties.

Reply:
thanks for the detailed explanation @mliao95 👌